### ISSUE TITLE
godot 4.2 tangetns warnings

### ISSUE THREAD

**Issue by hakur** [TokisanGames/Terrain3D#268]
### Terrain3D version

v0.8.4-alpha_gd4.1.3

### Godot version

v4.2.stable.official.46dc27791

### System information

windows10/64-Vulkan Mobile() - dedicated AMD 6700XT

### Is the issue reproducable in the demo?

Yes

## what happend
godot output warning log, get stuck for serveral seconds when run game
```
WARNING: Attempting to use a shader that requires tangents with a mesh that doesn't contain tangents. Ensure that meshes are imported with the 'ensure_tangents' option. If creating your own meshes, add an `ARRAY_TANGENT` array (when using ArrayMesh) or call `generate_tangents()` (when using SurfaceTool).
     at: _geometry_instance_add_surface_with_material (servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp:3725)
WARNING: Attempting to use a shader that requires tangents with a mesh that doesn't contain tangents. Ensure that meshes are imported with the 'ensure_tangents' option. If creating your own meshes, add an `ARRAY_TANGENT` array (when using ArrayMesh) or call `generate_tangents()` (when using SurfaceTool).
     at: _geometry_instance_add_surface_with_material (servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp:3725)
```
## except
compatible with 4.2, and no tangents warning in console
## how to reproduce
1. create new project
2. copy and enable terrain3d addon 
3. create new 3d scene
4. add Terrain3D node
5. see console output many tangents warning
https://github.com/TokisanGames/Terrain3D/assets/12321972/62833961-ed5e-4289-b07e-644c78bc3766

### debug logging:
```log
Terrain3D::_notification: NOTIFICATION_ENTER_WORLD
Terrain3D::_notification: NOTIFICATION_ENTER_TREE
Terrain3D::_initialize: Checking material, storage, texture_list, signal, and mesh initialization
Terrain3D::_initialize: Creating blank material
Terrain3D::_initialize: Creating blank storage
Terrain3DStorage::set_version: 0.842
Terrain3D::_initialize: Creating blank texture list
Terrain3D::_initialize: Connecting texture_list.textures_changed to _material->_update_texture_arrays()
Terrain3D::_initialize: Connecting region_size_changed signal to _material->_set_region_size()
Terrain3D::_initialize: Connecting regions_changed signal to _material->_update_regions()
Terrain3D::_initialize: Connecting height_maps_changed signal to update_aabbs()
Terrain3DMaterial::initialize: Initializing material
Terrain3DMaterial::_preload_shaders: Loaded shader insert: world_noise
Terrain3DMaterial::_preload_shaders: Loaded shader insert: WORLD_NOISE1
Terrain3DMaterial::_preload_shaders: Loaded shader insert: WORLD_NOISE2
Terrain3DMaterial::_preload_shaders: Loaded shader insert: debug_views
Terrain3DMaterial::_preload_shaders: Loaded shader insert: DEBUG_CHECKERED
Terrain3DMaterial::_preload_shaders: Loaded shader insert: DEBUG_GREY
Terrain3DMaterial::_preload_shaders: Loaded shader insert: DEBUG_HEIGHTMAP
Terrain3DMaterial::_preload_shaders: Loaded shader insert: DEBUG_COLORMAP
Terrain3DMaterial::_preload_shaders: Loaded shader insert: DEBUG_ROUGHMAP
Terrain3DMaterial::_preload_shaders: Loaded shader insert: DEBUG_CONTROL_TEXTURE
Terrain3DMaterial::_preload_shaders: Loaded shader insert: DEBUG_CONTROL_BLEND
Terrain3DMaterial::_preload_shaders: Loaded shader insert: DEBUG_TEXTURE_HEIGHT
Terrain3DMaterial::_preload_shaders: Loaded shader insert: DEBUG_TEXTURE_NORMAL
Terrain3DMaterial::_preload_shaders: Loaded shader insert: DEBUG_TEXTURE_ROUGHNESS
Terrain3DMaterial::_preload_shaders: Loaded shader insert: DEBUG_VERTEX_GRID
Terrain3DMaterial::_preload_shaders: Loaded shader insert: main
Terrain3DMaterial::_set_region_size: Setting region size in material: 1024
Terrain3DMaterial::initialize: Mat RID: RID(750764578308461), _shader RID: RID(750768873275431)
Terrain3DMaterial::_update_shader: Updating shader
Terrain3DMaterial::_generate_shader_code: Generating default shader code
Terrain3DMaterial::_update_shader: Mat rid: RID(750764578308461), _shader rid: RID(750768873275431)
Terrain3DMaterial::_update_shader: _active_params: [&"height_blending", &"blend_sharpness", &"noise_scale", &"noise_texture"]
Terrain3DUtil::print_dict: Printing Dictionary: _shader_params
Terrain3DUtil::print_dict: Key: height_blending, Value: true
Terrain3DUtil::print_dict: Key: blend_sharpness, Value: 0.87000000476837
Terrain3DUtil::print_dict: Key: noise_scale, Value: 0.10000000149012
Terrain3DUtil::print_dict: Key: noise_texture, Value: <null>
Terrain3DMaterial::_update_shader: Generating default noise_texture for shader
Terrain3DMaterial::_update_regions: Updating region maps in shader
Terrain3DMaterial::_update_regions: Height map RID: RID(0)
Terrain3DMaterial::_update_regions: Control map RID: RID(0)
Terrain3DMaterial::_update_regions: Color map RID: RID(0)
Terrain3DMaterial::_update_regions: _region_map.size(): 256
Terrain3DMaterial::_update_regions: Region map
Terrain3DMaterial::_update_regions: Region_offsets size: 0 []
Terrain3DMaterial::_generate_region_blend_map: Regenerating (512, 512) region blend map
Terrain3DGeneratedTex::create: RenderingServer creating Texture2D
Terrain3DUtil::dump_gen: Generated blend_map RID: RID(750811822952070), dirty: false, image: <Image#-9223369688195468795>
Terrain3DTextureList::update_list: Reconnecting texture signals
Terrain3DMaterial::_update_texture_arrays: Updating texture arrays in shader
Terrain3DMaterial::set_show_checkered: Enable set_show_checkered: true
Terrain3DMaterial::_update_shader: Updating shader
Terrain3DMaterial::_generate_shader_code: Generating default shader code
Terrain3DMaterial::_update_shader: Mat rid: RID(750764578308461), _shader rid: RID(750768873275431)
Terrain3DMaterial::_update_shader: _active_params: [&"height_blending", &"blend_sharpness", &"noise_scale", &"noise_texture"]
Terrain3DUtil::print_dict: Printing Dictionary: _shader_params
Terrain3DUtil::print_dict: Key: height_blending, Value: true
Terrain3DUtil::print_dict: Key: blend_sharpness, Value: 0.87000000476837
Terrain3DUtil::print_dict: Key: noise_scale, Value: 0.10000000149012
Terrain3DUtil::print_dict: Key: noise_texture, Value: <NoiseTexture2D#-9223369688245799780>
Terrain3DMaterial::_update_texture_arrays: No textures, enabling checkered view
Terrain3D::_build: Building the terrain meshes
Terrain3DGeoClipMap::generate: Generating meshes of size: 48 levels: 7
Terrain3DGeoClipMap::_create_mesh: Creating mesh via the Rendering server
Terrain3DGeoClipMap::_create_mesh: Setting custom aabb: (0, 0, 0), (49, 0.1, 49)
Terrain3DGeoClipMap::_create_mesh: Creating mesh via the Rendering server
Terrain3DGeoClipMap::_create_mesh: Setting custom aabb: (-96, 0, -96), (193, 0.1, 193)
Terrain3DGeoClipMap::_create_mesh: Creating mesh via the Rendering server
Terrain3DGeoClipMap::_create_mesh: Setting custom aabb: (-97.5, 0, -97.5), (194.5, 0.1, 194.5)
Terrain3DGeoClipMap::_create_mesh: Creating mesh via the Rendering server
Terrain3DGeoClipMap::_create_mesh: Setting custom aabb: (-97.5, 0, -97.5), (194.5, 0.1, 194.5)
Terrain3DGeoClipMap::_create_mesh: Creating mesh via the Rendering server
Terrain3DGeoClipMap::_create_mesh: Setting custom aabb: (-97.5, 0, -97.5), (291.5, 0.1, 291.5)
Terrain3D::_build: Creating mesh instances
Terrain3D::_notification: NOTIFICATION_READY
Terrain3D::_initialize: Checking material, storage, texture_list, signal, and mesh initialization
Terrain3D::set_plugin: Received editor plugin: @EditorPlugin@17257:<EditorPlugin#1503171503348>
Terrain3D::__process: camera is null, getting the current one
Terrain3D::_find_cameras: Found a Camera3D at: /root/@EditorNode@17120/@Panel@13/@VBoxContainer@14/@HSplitContainer@17/@HSplitContainer@25/@HSplitContainer@33/@VBoxContainer@34/@VSplitContainer@36/@VSplitContainer@62/@VBoxContainer@63/@PanelContainer@110/MainScreen/@Node3DEditor@10009/@HSplitContainer@9519/@HSplitContainer@9521/@VSplitContainer@9523/@Node3DEditorViewportContainer@9524/@Node3DEditorViewport@9563/@SubViewportContainer@9525/@SubViewport@9526/@Camera3D@9528
Terrain3D::_find_cameras: Found a Camera3D at: /root/@EditorNode@17120/@Panel@13/@VBoxContainer@14/@HSplitContainer@17/@HSplitContainer@25/@HSplitContainer@33/@VBoxContainer@34/@VSplitContainer@36/@VSplitContainer@62/@VBoxContainer@63/@PanelContainer@110/MainScreen/@Node3DEditor@10009/@HSplitContainer@9519/@HSplitContainer@9521/@VSplitContainer@9523/@Node3DEditorViewportContainer@9524/@Node3DEditorViewport@9602/@SubViewportContainer@9564/@SubViewport@9565/@Camera3D@9567
Terrain3D::_find_cameras: Found a Camera3D at: /root/@EditorNode@17120/@Panel@13/@VBoxContainer@14/@HSplitContainer@17/@HSplitContainer@25/@HSplitContainer@33/@VBoxContainer@34/@VSplitContainer@36/@VSplitContainer@62/@VBoxContainer@63/@PanelContainer@110/MainScreen/@Node3DEditor@10009/@HSplitContainer@9519/@HSplitContainer@9521/@VSplitContainer@9523/@Node3DEditorViewportContainer@9524/@Node3DEditorViewport@9641/@SubViewportContainer@9603/@SubViewport@9604/@Camera3D@9606
Terrain3D::_find_cameras: Found a Camera3D at: /root/@EditorNode@17120/@Panel@13/@VBoxContainer@14/@HSplitContainer@17/@HSplitContainer@25/@HSplitContainer@33/@VBoxContainer@34/@VSplitContainer@36/@VSplitContainer@62/@VBoxContainer@63/@PanelContainer@110/MainScreen/@Node3DEditor@10009/@HSplitContainer@9519/@HSplitContainer@9521/@VSplitContainer@9523/@Node3DEditorViewportContainer@9524/@Node3DEditorViewport@9680/@SubViewportContainer@9642/@SubViewport@9643/@Camera3D@9645
Terrain3D::_grab_camera: Connecting to the first editor camera
```
```log
  servers/rendering/renderer_rd/forward_mobile/render_forward_mobile.cpp:2525 - Attempting to use a shader that requires tangents with a mesh that doesn't contain tangents. Ensure that meshes are imported with the 'ensure_tangents' option. If creating your own meshes, add an `ARRAY_TANGENT` array (when using ArrayMesh) or call `generate_tangents()` (when using SurfaceTool). (User)
  servers/rendering/renderer_rd/forward_mobile/render_forward_mobile.cpp:2525 - Attempting to use a shader that requires tangents with a mesh that doesn't contain tangents. Ensure that meshes are imported with the 'ensure_tangents' option. If creating your own meshes, add an `ARRAY_TANGENT` array (when using ArrayMesh) or call `generate_tangents()` (when using SurfaceTool). (User)
  servers/rendering/renderer_rd/forward_mobile/render_forward_mobile.cpp:2525 - Attempting to use a shader that requires tangents with a mesh that doesn't contain tangents. Ensure that meshes are imported with the 'ensure_tangents' option. If creating your own meshes, add an `ARRAY_TANGENT` array (when using ArrayMesh) or call `generate_tangents()` (when using SurfaceTool). (User)
```

**Comment by TokisanGames** [comment]
Fixed in fc2f60da66713bdac2b8a209ef3ba6e50c20df1d.

Use the version for Godot 4.2 linked in the [release notes](https://github.com/TokisanGames/Terrain3D/releases/tag/v0.8.4-alpha_gd4.1.3).

**Comment by hakur** [comment]
I got, thanks for your answer
